app/ts/simulation/priceEstimator.ts

```diff
@@ -21,99 +22,96 @@ export class PriceEstimator {
 		this.ethereum = ethereum
 	}
 
-	public async estimateEthereumPricesForTokens(tokens: TokenDecimals[]) : Promise<TokenPriceEstimate[]> {
+	public async estimateEthereumPricesForTokens(tokens: TokenDecimals[], quote?: TokenDecimals) : Promise<TokenPriceEstimate[]> {
 		if (tokens.length == 0) return []
 
-		const chainId = await this.ethereum.getChainId()
+		const chainId = this.ethereum.getChainId()
 		const chainString = chainId.toString()
 		if (!isSupportedChain(chainString)) return []
+		// Support only mainnet for now
+		// TODO: fix this
+		if (chainId !== 1n) return []
 
-		const amountOutMin = 0n
-		const sender = CHAINS[chainString].eth_donator
-		const block = await this.ethereum.getBlock()
-		const deadline = BigInt( block.timestamp.getTime() + 1000 * 1000)
-
-		const transactionCount = await this.ethereum.getTransactionCount(sender)
-		let inOutResults: TokenPriceEstimate[] = []
-		const swapInterface = new ethers.Interface(ABI)
-		for ( const token of tokens) {
-			if ( token.token === CHAINS[chainString].weth ) {
-				inOutResults.push({
-					token: addressString(CHAINS[chainString].weth),
-					inOutAmount: [10n ** 18n, 10n ** 18n],
-					decimals: 18n,
+		const quoteToken = quote ?? {
+			address: BigInt(CHAINS[chainString].weth),
```

Comment:
Would we use `weth` as the quote token for other chains as well (e.g., Optimism/Arbitrum/BNB)?  I wonder `CHAINS` should contain a `quoteToken` so we can define a different quote token for each chain?

Reply:
currently this `weth´ needs to correspond something that ~equals the chains main coin. I feel the strategy would be to always quote in this, and then add a separate token that we would swap to users preference token, which would be a separate setting. I feel this change should not be part of this PR but another one